homework/blackjack/tests/dealerTest.py

```diff
@@ -0,0 +1,124 @@
+from dealer import Dealer
+from card import Card
+import unittest
+import xprint
+from xprint import xPrint
+
+
+class DealerTest(unittest.TestCase):
+    """
+    Test out the functionality of class Dealer
+    """
+
+    def testInvalidDeck(self):
+        """
+        Dealer should raise 
+        ValueError
+        if the numerOfDecks is less than 1
+        """
+        deck = 0
+        self.assertRaises(ValueError, Dealer, deck)
+
+    def __validateDeckCompleteness(self, cards):
+        """
+        Check if a deck of card contains 52 cards,
+        13 for each suit.
+        """
+        cardsDict = {
+            "CLUBS": set(),
+            "DIAMONDS": set(),
+            "HEARTS": set(),
+            "SPADES": set(),
+        }
+        for card in cards:
+            self.assertIn(card.getSuit(), cardsDict)
+            self.assertNotIn(card.getRank(), cardsDict[card.getSuit()])
+            cardsDict[card.getSuit()].add(card.getRank())
+        #xprint.debugPrint = True
+        xPrint(cardsDict)
+
+    def testCreateADeckOfCard(self):
```

Comment:
Test that a deck of cards is valid e.g. 1 ace of spaces, 1 queen of hearts etc..?